addons/api/addon/serializers/host-catalog.js

```diff
@@ -34,7 +34,7 @@ export default class HostCatalogSerializer extends ApplicationSerializer {
     ) {
       if (options.isNestedAttribute) {
         json.attributes[key] = null;
-      } else if (options.isNestedSecret && json.secrets[key]) {
+      } else if (options.isNestedSecret && json.secrets && json.secrets[key]) {
```

Comment:
Would this have worked?

```suggestion
      } else if (options.isNestedSecret && json?.secrets[key]) {
```

Reply:
thank u for the suggestion, this made me realize that something similar works.

Reply:
Woops I had it on the wrong field 😅